### PR TITLE
podio, edm4hep: fix tests

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
LGTM

File: var/spack/repos/builtin/packages/edm4hep/package.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def cmake_args(self):
         # C++ Standard
         args.append(self.define('CMAKE_CXX_STANDARD',
                     self.spec.variants['cxxstd'].value))
+        args.append("-DBUILD_TESTING=%s" % self.run_tests)
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/edm4hep/package.py]:
```suggestion
        args.append(self.define("BUILD_TESTING", self.run_tests))
```

File: var/spack/repos/builtin/packages/podio/package.py [hunk 2]
```diff
@@ -44,12 +44,14 @@ class Podio(CMakePackage):
...
     depends_on('sio', type=('build', 'link'), when='+sio')
+    depends_on('catch2@3.0.1:', type=('test'), when="@0.13:")
 
     conflicts('+sio', when='@:0.12', msg='sio support requires at least podio@0.13')
 
     def cmake_args(self):
         args = [
-            self.define_from_variant('ENABLE_SIO', 'sio')
+            self.define_from_variant('ENABLE_SIO', 'sio'),
+            "-DBUILD_TESTING=%s" % self.run_tests,
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/podio/package.py]:
```suggestion
            self.define("BUILD_TESTING", self.run_tests),
```

**Review by sethrj — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**